forte/data/entry_converter.py

```diff
@@ -0,0 +1,209 @@
+                type_name=entry.entry_type(),
+                image_payload_idx=entry.image_payload_idx,
+                tid=entry.tid,
+            )
+        elif isinstance(entry, Grids):
+            data_store_ref.add_grid_raw(
+                type_name=entry.entry_type(),
+                image_payload_idx=entry.image_payload_idx,
+                tid=entry.tid,
+            )
+        elif isinstance(entry, MultiPackLink):
+            data_store_ref.add_multipack_link_raw(
+                type_name=entry.entry_type(),
+                parent_pack_id=entry.parent[0],
+                parent_tid=entry.parent[1],
+                child_pack_id=entry.child[0],
+                child_tid=entry.child[1],
+                tid=entry.tid,
+            )
+        elif isinstance(entry, MultiPackGroup):
+            data_store_ref.add_multipack_group_raw(
+                type_name=entry.entry_type(),
+                member_type=get_full_module_name(entry.MemberType),
+                tid=entry.tid,
+            )
+        elif isinstance(entry, MultiPackGeneric):
+            data_store_ref.add_multipack_generic_raw(
+                type_name=entry.entry_type(),
+                tid=entry.tid,
+            )
+        else:
+            raise ValueError(
+                f"Invalid entry type {type(entry)}. A valid entry "
+                f"should be an instance of Annotation, Link, Group, Generics "
+                "or AudioAnnotation."
+            )
+
+        # Store all the dataclass attributes to DataStore
+        for attribute in entry.__dataclass_fields__:
+            value = getattr(entry, attribute, None)
+            if not value:
+                continue
+            if isinstance(value, Entry):
+                value = value.tid
+            elif isinstance(value, FDict):
+                value = {key: val.tid for key, val in value.items()}
+            elif isinstance(value, FList):
+                value = [val.tid for val in value]
+            data_store_ref.set_attribute(
+                tid=entry.tid, attr_name=attribute, attr_value=value
+            )
+
+        # Cache the stored entry and its tid
+        self._entry_dict[entry.tid] = entry
+
+    def get_entry_object(self, tid: int, pack: PackType) -> EntryType:
+        """
+        Convert a tid to its corresponding entry object.
+        """
+
+        # Check if the tid is cached
+        if tid in self._entry_dict:
+            return self._entry_dict[tid]  # type: ignore
+
+        data_store_ref = pack._data_store  # pylint: disable=protected-access
+        entry_data, entry_type = data_store_ref.get_entry(tid=tid)
+        entry_class = get_class(entry_type)
+        entry: Entry
+        # Here the entry arguments are optional (begin, end, parent, ...) and
+        # the value can be arbitrary since they will all be routed to DataStore.
+        if issubclass(entry_class, (Annotation, AudioAnnotation)):
```

Comment:
similarly, let's not do `issubclass` check here. In general, let's remove all the class check as much as we can.

Reply:
https://github.com/asyml/forte/blob/master/forte/data/data_store.py#L762